tasks/argocd/src/main/java/com/walmartlabs/concord/plugins/argocd/CallRetry.java

```diff
@@ -23,23 +23,24 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import java.util.Collection;
-import java.util.Collections;
-import java.util.Optional;
+import java.util.*;
 import java.util.concurrent.Callable;
 
 public class CallRetry<R> {
     private static final Logger log = LoggerFactory.getLogger(CallRetry.class);
     private final Callable<R> mainAttempt;
     private final Collection<Callable<Optional<R>>> fallbackAttempts;
+    private final List<Class<? extends Exception>> exceptionsToNotRetry;
```

Comment:
Prefer a `Set` so entries are not repeated.

Reply:
Updated with set